java/pattern/constraint/TypeConstraint.java

```diff
@@ -466,7 +466,22 @@ public boolean isOwns() {
 
         @Override
         public String toString() {
-            return OWNS.toString() + SPACE + attributeType + (isKey ? "" + SPACE + IS_KEY : "");
+            StringBuilder syntax = new StringBuilder();
+            syntax.append(OWNS).append(SPACE);
+
+            if (!attributeType.label().isPresent()) syntax.append(attributeType);
+            else syntax.append(attributeType.label().get().label());
+
+            if (isKey) syntax.append(SPACE).append(IS_KEY);
+
+            if (overriddenAttributeType!=null) {
+                syntax.append(SPACE).append(AS).append(SPACE);
+
+                if (!overriddenAttributeType.label().isPresent()) syntax.append(overriddenAttributeType);
+                else syntax.append(overriddenAttributeType.label().get().label());
```

Comment:
Same as above: this logic is not needed given that `TypeVariable.toString()` already takes care of this.